Formula/sratoolkit.rb

```diff
@@ -38,36 +38,36 @@ class Sratoolkit < Formula
     end
   end
 
+  depends_on "bison" => :build
   depends_on "cmake" => :build
+  depends_on "doxygen" => :build
+
+  depends_on "gcc"
   depends_on "hdf5"
-  depends_on macos: :catalina
 
+  uses_from_macos "flex" => :build
   uses_from_macos "libxml2"
 
-  # Modify cmake scripts to avoid building tests when BUILD_TESTING=OFF and
-  # to install into CMAKE_INSTALL_LIBDIR rather than CMAKE_INSTALL_PREFIX/lib64.
-  # Issue ref: https://github.com/ncbi/sra-tools/issues/638
-  # Issue ref: https://github.com/ncbi/sra-tools/issues/639
   patch :DATA
 
   def install
+    ENV.cxx11
+
     (buildpath/"ncbi-vdb-source").install resource("ncbi-vdb")
 
     # Workaround to allow clang/aarch64 build to use the gcc/arm64 directory
     # Issue ref: https://github.com/ncbi/ncbi-vdb/issues/65
     ln_s "../gcc/arm64", buildpath/"ncbi-vdb-source/interfaces/cc/clang/aarch64" if Hardware::CPU.arm?
```

Comment:
This is previous workaround I added to allow building ARM64 with clang. Upstream updated directory name to use `arm64` now - https://github.com/ncbi/ncbi-vdb/commit/01848812a06ac35528487221114d643402ec02b6

If you want to use `gcc` instead, then it should be restricted to ARM64 and should use compiler selection logic instead (`fails_with :clang`). Directly setting `CC`/`CXX` will skip the shims.

Reply:
I don't know what we need to go down the `gcc` route if there is a simpler way to do this. Without the `gcc` commit we are still getting this:

```
fatal error: 'atomic32.h' file not found
#include <atomic32.h>
         ^~~~~~~~~~~~
1 error generated.
```

Reply:
Clang workaround would be:
```diff
@@ -41,8 +41,6 @@ class Sratoolkit < Formula
   depends_on "bison" => :build
   depends_on "cmake" => :build
   depends_on "doxygen" => :build
-
-  depends_on "gcc"
   depends_on "hdf5"
 
   uses_from_macos "flex" => :build
@@ -57,17 +55,12 @@ class Sratoolkit < Formula
 
     # Workaround to allow clang/aarch64 build to use the gcc/arm64 directory
     # Issue ref: https://github.com/ncbi/ncbi-vdb/issues/65
-    ln_s "../gcc/arm64", buildpath/"ncbi-vdb-source/interfaces/cc/clang/aarch64" if Hardware::CPU.arm?
+    ln_s "../gcc/arm64", buildpath/"ncbi-vdb-source/interfaces/cc/clang/arm64" if Hardware::CPU.arm?
 
     # Need to use HDF 1.10 API: error: too few arguments to function call, expected 5, have 4
     # herr_t h5e = H5Oget_info_by_name( self->hdf5_handle, buffer, &obj_info, H5P_DEFAULT );
     ENV.append_to_cflags "-DH5_USE_110_API"
 
-    gcc = Formula["gcc"]
-    gcc_major_ver = gcc.any_installed_version.major
-    ENV["CC"] = Formula["gcc"].opt_bin/"gcc-#{gcc_major_ver}"
-    ENV["CXX"] = Formula["gcc"].opt_bin/"g++-#{gcc_major_ver}"
-
     system "cmake", "-S", "ncbi-vdb-source", "-B", "ncbi-vdb-build", *std_cmake_args,
                     "-DNGS_INCDIR=#{buildpath}/ngs/ngs-sdk"
     system "cmake", "--build", "ncbi-vdb-build"
```

GCC usage with using compiler selection DSL would be:
```diff
@@ -41,13 +41,18 @@ class Sratoolkit < Formula
   depends_on "bison" => :build
   depends_on "cmake" => :build
   depends_on "doxygen" => :build
-
-  depends_on "gcc"
   depends_on "hdf5"
 
   uses_from_macos "flex" => :build
   uses_from_macos "libxml2"
 
+  on_macos do
+    on_arm do
+      depends_on "gcc"
+      fails_with :clang
+    end
+  end
+
   patch :DATA
 
   def install
@@ -55,19 +60,10 @@ class Sratoolkit < Formula
 
     (buildpath/"ncbi-vdb-source").install resource("ncbi-vdb")
 
-    # Workaround to allow clang/aarch64 build to use the gcc/arm64 directory
-    # Issue ref: https://github.com/ncbi/ncbi-vdb/issues/65
-    ln_s "../gcc/arm64", buildpath/"ncbi-vdb-source/interfaces/cc/clang/aarch64" if Hardware::CPU.arm?
-
     # Need to use HDF 1.10 API: error: too few arguments to function call, expected 5, have 4
     # herr_t h5e = H5Oget_info_by_name( self->hdf5_handle, buffer, &obj_info, H5P_DEFAULT );
     ENV.append_to_cflags "-DH5_USE_110_API"
 
-    gcc = Formula["gcc"]
-    gcc_major_ver = gcc.any_installed_version.major
-    ENV["CC"] = Formula["gcc"].opt_bin/"gcc-#{gcc_major_ver}"
-    ENV["CXX"] = Formula["gcc"].opt_bin/"g++-#{gcc_major_ver}"
-
     system "cmake", "-S", "ncbi-vdb-source", "-B", "ncbi-vdb-build", *std_cmake_args,
                     "-DNGS_INCDIR=#{buildpath}/ngs/ngs-sdk"
     system "cmake", "--build", "ncbi-vdb-build"
```

Reply:
Looks like updating the workaround fixed us on macOS, but now Linux is failing.